src/server/worker/datum/iterator.go

```diff
@@ -118,11 +118,15 @@ func iterateMeta(pachClient *client.APIClient, commit *pfs.Commit, pathRange *pf
 			if pfsserver.IsFileNotFoundErr(err) || errors.Is(err, io.EOF) {
 				return nil
 			}
-			return errors.EnsureStack(err)
+			return errors.Wrap(err, "next")
 		}
 		meta := &Meta{}
-		if err := jsonpb.Unmarshal(tr, meta); err != nil {
```

Comment:
Leaving a note to come back to this.  seems a little sus, like we went from unmarshaling line by line, to reading the whole thing into memory.

Reply:
Circling back:  The outer loop is a tar stream, and you're only reading a file at a time.  We should really have a max on this but, I'm sure we are using `io.ReadAll` elsewhere.

Reply:
`jsonpb.Unmarshal(io.Reader, ...)` is a weird function; it basically uses a json.Decoder to read the first message into memory, and then it unmarshals that, and then returns.  I'm not sure those were the semantics we wanted here, so I used `protojson.Unmarshal` instead of `protoutil.NewProtoJSONDecoder().UnmarshalNext()`.  Maybe that's a mistake?